common4j/src/main/com/microsoft/identity/common/java/util/ported/DataBag.java

```diff
@@ -0,0 +1,40 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.util.ported;
+
+import com.microsoft.identity.common.java.interfaces.INameValueStorage;
+
+import java.io.Serializable;
+
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+@Getter
+@Accessors(prefix = "m")
+public class DataBag {
+    private final INameValueStorage<Boolean> mBooleanMap = new MapWithDefault<>();
+    private final INameValueStorage<byte[]> mByteArrayMap = new MapWithDefault<>();
+    private final INameValueStorage<String> mStringMap = new MapWithDefault<>();
+    private final INameValueStorage<Integer> mIntMap = new MapWithDefault<>();
+    private final INameValueStorage<Serializable> mSerializableMap = new MapWithDefault<>();
+}
```

Comment:
Can we put the following utility methods on this class?

```java
void put(String key, Boolean value) {
   mBooleanMap.put(key, value);
}

void put(String key, byte[] value) {
   mByteArrayMap.put(key, value);
}

// do the same for all the other maps as well

// also provider a getter as follows:
Object get(String key) {
 // does a look-up against each of these and returns the right value
}
```

Reply:
Also see comment here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1475/files#r674202035

Reply:
Another thought is that we may able to simplify this even further by having just one map for byte[]

```java
private final INameValueStorage<byte[]> mByteArrayMap = new MapWithDefault<>();
```

This is because all the types here are Serializable anyway...so we should be able to convert them to bytes

Reply:
I could create those helper function (that's actually my original design - it's just that it make this classes cluttered, since we'll need get() put() and getOrDefault() for each of them, 3*5 = 15 methods in this class)

That said, with one map, it's better to clutter thing here than in DataBagUtil, so i'll make the change.

Reply:
(what you suggested is actually similar to how Bundle is implemented)

Reply:
actually i can just use generic getters.